MarkBind/markbind
# Refactor cli/index.js

**What is the purpose of this pull request?**




**Overview of changes:**

Refactors `cli/index.js` into multiple files, and separates it by command (`init`, `serve`, `build`, `deploy`).

**Anything you'd like to highlight/discuss:**

This refactor will encourage further modularity of our CLI and allow for easier development / debugging of the CLI commands and processes. The idea to do this was initially proposed in [my comment](https://github.com/MarkBind/markbind/issues/1756#issuecomment-1466383919) in #1756 to aid in the refactoring of `Site/index.js`, and was again mentioned in #2218.

A further refactor, specifically for the `serve` command, may be a possibility as well.

And of course, we should add more tests for each of the commands (in a separate PR).

**Testing instructions:**

Ensure that all the commands - `init`, `serve`, `build`, and `deploy` - are working as intended and as specified in the User Guide.



**Proposed commit message: (wrap lines at 72 characters)**


```
Refactor cli/index.js

The code for cli/index.js, especially for the serve command, is
becoming longer.

As more functionalities are added to the CLI commands, it is
becoming difficult to navigate the code.

Extracting the code for each command into a separate function
encourages modularity in developing the commands without 
worrying about coupling, allowing for a better debugging
experience and better code quality in general.

Let's extract all the code bodies for each command into a
separate file dedicated to a single command, and place all
common functions into the cli/src/util directory.

Doing this is preferable as we can prevent cli/index.js from
becoming too lengthy for a simple refactoring like Site/index.js.
```
---

**Checklist:** :ballot_box_with_check:
